README.md

```diff
@@ -1,13 +1,161 @@
+  * SignNow uses Authorization: Basic <credentials>. To get your access token, have your client ID and secret base-64 encoded first
+
+
+## <a name="platform-dependencies"></a>Platform Dependencies
+#### Windows
+  * .Net Framework 4.5 or newer version should be installed in your system, or
+  * .Net Standard 1.2 or 2.0, or
+  * .Net Core 2.2 and newer
+
+#### MacOS and Linux
+  * .Net Core 2.2 and newer
+
+
+## <a name="xml-doc-generation"></a>XML doc generation
+
+For XML documentation generation, install InheritDocTool (the project build will fail without it):
+
+```bash
 dotnet tool install -g InheritDocTool
 ```
 
-You can see details about InheritDoc [here](https://www.inheritdoc.io)
+More about the InheritDoc [here](https://www.inheritdoc.io)
+
+
+## <a name="installation"></a>Installation
+
+### Install .Net package
+
+```bash
+dotnet add package SignNow.Net --version <package-version>
+```
+
+### Install using Package manager
+
+```powershell
+Install-Package SignNow.Net -Version <package-version>
+```
+
+## <a name="documentation"></a>Documentation
+
+Read about the available SignNow features in [SignNow API Docs](https://github.com/signnow/SignNow.NET.wiki.git) .
+
+
+## <a name="features"></a>Features
+
+### <a name="authorization"></a> Authorization
+
+Get your access token via OAuth 2.0 service.
+
+```csharp
+string clientId = "0fa****-EXAMPLE_CLIENT_ID-****13";
+string clientSecret = "0fb**-EXAMPLE_CLIENT_SECRET-**13";
+
+string userLogin = "signnow_dotnet_sdk@example.com";
+string userPassword = "example-user-password";
+
+var oauth = new OAuth2Service(clientId, clientSecret);
+
+var token = oauth.GetTokenAsync(userLogin, userPassword, Scope.All).Result;
+}
+```
+
+### <a name="upload-document"></a> Upload a document to SignNow
+
+All the features in SignNow require a `document_id`. Once you upload a document to SignNow, you get the `document_id` from a successful response.
+```csharp
+string documentId = default;
+
+var pdfFilePath = "./file-path/document-example.pdf";
+var pdfFileName = "document-example.pdf";
+
+/// using token from the Authorization step
+var documentService = new DocumentService(token);
+
+using (var fileStream = File.OpenRead(pdfFilePath))
+{
+    var uploadResponse = documentService.UploadDocumentAsync(fileStream, pdfFileName, default).Result;
+
+    documentId = uploadResponse.Id;
+}
+```
+
+### <a name="download-document"></a> Download a document from SignNow
+
+Choose the type of download for your document:
+* `PdfOriginal` - download a document in a state it's been when uploaded to SignNow, before any changes
+* `PdfCollapsed` - download a document in PDF file format
+* `ZipCollapsed` - download a document in ZIP archive
+* `PdfWithHistory` - download a document with its history, a full log of changes on a separate page.
+
+```csharp
+/// using `documentId` from the Upload document step
+
+var downloadPdf = documentService.DownloadDocumentAsync(documentId, downloadType.PdfCollapsed).Result;
```

Comment:
need details how to actually save the file.

Reply:
added